crates/proof-of-sql/src/sql/proof_exprs/comparison_util.rs

```diff
@@ -31,6 +31,70 @@ fn unchecked_subtract_impl<'a, S: Scalar>(
     Ok(result)
 }
 
+#[allow(
+    clippy::missing_panics_doc,
+    reason = "precision and scale are validated prior to calling this function, ensuring no panic occurs"
+)]
+/// Scale LHS and RHS to the same scale if at least one of them is decimal
+/// and take the difference. This function is used for comparisons.
```

Comment:
You should only use `allow(clippy::missing_panics_doc)` if the function is guarenteed to not panic. If invalid parameters are passed, then it will/should panic, so we should document that.
```suggestion
/// Scale LHS and RHS to the same scale if at least one of them is decimal
/// and take the difference. This function is used for comparisons.
///
/// # Panics
/// The precision and scale passed must be valid. A panic may occur if they are not.
```